libbeat/outputs/tls.go

```diff
@@ -53,167 +84,217 @@ func (c *TLSConfig) Validate() error {
 		return ErrKeyNoCertificate
 	}
 
-	if _, err := parseTLSVersion(c.MinVersion); err != nil {
-		return err
-	}
-	if _, err := parseTLSVersion(c.MaxVersion); err != nil {
-		return err
-	}
-
-	if _, err := parseTLSCipherSuites(c.CipherSuites); err != nil {
-		return err
-	}
-
-	if _, err := parseCurveTypes(c.CurveTypes); err != nil {
-		return err
-	}
-
 	return nil
 }
 
+func (c *TLSConfig) IsEnabled() bool {
+	return c != nil && (c.Enabled == nil || *c.Enabled)
+}
+
 // LoadTLSConfig will load a certificate from config with all TLS based keys
 // defined. If Certificate and CertificateKey are configured, client authentication
 // will be configured. If no CAs are configured, the host CA will be used by go
 // built-in TLS support.
-func LoadTLSConfig(config *TLSConfig) (*tls.Config, error) {
-	if config == nil {
+func LoadTLSConfig(config *TLSConfig) (*transport.TLSConfig, error) {
+	if !config.IsEnabled() {
 		return nil, nil
 	}
 
+	fail := multierror.Errors{}
+	logFail := func(es ...error) {
+		for _, e := range es {
+			if e != nil {
+				fail = append(fail, e)
+			}
+		}
+	}
+
+	var cipherSuites []uint16
+	for _, suite := range config.CipherSuites {
+		cipherSuites = append(cipherSuites, uint16(suite))
+	}
+
+	var curves []tls.CurveID
+	for _, id := range config.CurveTypes {
+		curves = append(curves, tls.CurveID(id))
+	}
+
+	cert, err := loadCertificate(&config.Certificate)
+	logFail(err)
+
+	cas, errs := loadCertificateAuthorities(config.CAs)
+	logFail(errs...)
+
+	// return config if no error occured
```

Comment:
Comment is kind of misleading, as I expected that in the next line, the config is returned.